app/models/teacher.rb

```diff
@@ -23,6 +23,7 @@ class Teacher < ApplicationRecord
   has_many :appropriate_bodies, through: :induction_periods
   has_many :events
   has_many :mentor_declarations, through: :mentor_training_periods, source: :declarations
+  has_many :ect_declarations, through: :ect_training_periods, source: :declarations
```

Comment:
Can we add a spec for this, and maybe update the `mentor_declarations` spec with `source(:declarations)`